gio/src/subclass/action.rs

```diff
@@ -0,0 +1,417 @@
+            from_glib_none(ret)
+        }
+    }
+
+    fn parent_enabled(&self) -> bool {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Action>() as *const ffi::GActionInterface;
+
+            let func = (*parent_iface)
+                .get_enabled
+                .expect("no parent \"get_enabled\" implementation");
+            let ret = func(self.obj().unsafe_cast_ref::<Action>().to_glib_none().0);
+            ret != 0
+        }
+    }
+
+    fn parent_state(&self) -> Option<glib::Variant> {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Action>() as *const ffi::GActionInterface;
+
+            let func = (*parent_iface)
+                .get_state
+                .expect("no parent \"get_state\" implementation");
+            let ret = func(self.obj().unsafe_cast_ref::<Action>().to_glib_none().0);
+            from_glib_none(ret)
+        }
+    }
+
+    fn parent_change_state(&self, value: glib::Variant) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Action>() as *const ffi::GActionInterface;
+
+            let func = (*parent_iface)
+                .change_state
+                .expect("no parent \"change_state\" implementation");
+            func(
+                self.obj().unsafe_cast_ref::<Action>().to_glib_none().0,
+                value.to_glib_none().0,
+            );
+        }
+    }
+
+    fn parent_activate(&self, parameter: Option<glib::Variant>) {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface =
+                type_data.as_ref().parent_interface::<Action>() as *const ffi::GActionInterface;
+
+            let func = (*parent_iface)
+                .activate
+                .expect("no parent \"activate\" implementation");
+            func(
+                self.obj().unsafe_cast_ref::<Action>().to_glib_none().0,
+                parameter.to_glib_none().0,
+            );
+        }
+    }
+
+    fn delegate_get_property(
+        &self,
+        prop_id: usize,
+        _pspec: &glib::ParamSpec,
+    ) -> Option<glib::Value> {
+        let type_: glib::Type = self.obj().type_();
+        let property = ActionProperty::from_type(type_, prop_id)?;
+        Some(match property {
+            ActionProperty::Name => self.name().to_value(),
+            ActionProperty::ParameterType => self.parameter_type().to_value(),
+            ActionProperty::StateType => self.state_type().to_value(),
+            ActionProperty::Enabled => self.is_enabled().to_value(),
+            ActionProperty::State => self.state().to_value(),
+        })
+    }
+
+    fn delegate_set_property(
```

Comment:
good call of adding this even if it just returns false; it would allow adding some attribute to glib::derived_properties to automatically use the `delegate_get_property` / `delegate_set_property` calls, simplifying things a bit.